server/base/src/main/java/org/apache/accumulo/server/manager/state/ZooTabletStateStore.java

```diff
@@ -67,50 +73,78 @@ public ClosableIterator<TabletManagement> iterator(List<Range> ranges,
+          ctx.getZooReaderWriter().getZooKeeper().getData(zpath, false, null);
+      final RootTabletMetadata rtm = new RootTabletMetadata(new String(rootTabletMetadata, UTF_8));
+      final SortedMapIterator iter = new SortedMapIterator(rtm.toKeyValues());
+      wri.init(iter, Map.of(), env);
+      tmi.init(wri,
+          Map.of(TabletManagementIterator.TABLET_GOAL_STATE_PARAMS_OPTION, parameters.serialize()),
+          env);
+      tmi.seek(new Range(), null, true);
+    } catch (KeeperException | InterruptedException | IOException e2) {
+      throw new IllegalStateException(
+          "Error setting up TabletManagementIterator for the root tablet", e2);
+    }
+
+    return new ClosableIterator<TabletManagement>() {
+
+      // We only want to return the root tablet metadata once
+      // and then hasNext should return false
+      private boolean returnedRootTabletMetadata = false;
 
       @Override
-      public boolean hasNext() {
-        return !finished;
+      public void close() {
+        closed.compareAndSet(false, true);
       }
 
       @Override
-      public TabletManagement next() {
-        finished = true;
-
-        final var actions = EnumSet.of(ManagementAction.NEEDS_LOCATION_UPDATE);
-        final TabletMetadata tm = ample.readTablet(RootTable.EXTENT, ReadConsistency.EVENTUAL);
-        String error = null;
-        try {
-          CompactionJobGenerator cjg =
-              new CompactionJobGenerator(new ServiceEnvironmentImpl(ctx), Map.of());
-          var jobs = cjg.generateJobs(tm,
-              EnumSet.of(CompactionKind.SYSTEM, CompactionKind.USER, CompactionKind.SELECTOR));
-          if (!jobs.isEmpty()) {
-            actions.add(ManagementAction.NEEDS_COMPACTING);
-          }
-        } catch (Exception e) {
-          log.error("Error computing tablet management actions for Root extent", e);
-          error = e.getMessage();
+      public boolean hasNext() {
+        if (closed.get()) {
+          return false;
         }
 
-        if (VolumeUtil.needsVolumeReplacement(parameters.getVolumeReplacements(), tm)) {
-          actions.add(ManagementAction.NEEDS_VOLUME_REPLACEMENT);
+        boolean result = tmi.hasTop() && !returnedRootTabletMetadata;
+        if (!result) {
+          close();
         }
-
-        return new TabletManagement(actions, tm, error);
-
+        return result;
       }
 
       @Override
-      public void remove() {
-        throw new UnsupportedOperationException();
-      }
+      public TabletManagement next() {
+        if (closed.get()) {
+          throw new NoSuchElementException(this.getClass().getSimpleName() + " is closed");
+        }
 
-      @Override
-      public void close() {}
+        Key k = tmi.getTopKey();
+        Value v = tmi.getTopValue();
+        Map<Key,Value> m = Map.of(k, v);
+        Entry<Key,Value> e = m.entrySet().iterator().next();
+        try {
+          TabletManagement tm = TabletManagementIterator.decode(e);
```

Comment:
Could drop `returnedRootTabletMetadata` variable if the following lines were added.

```suggestion
          tmi.next();
          Preconditions.checkState(!tmi.hasTop(), "Saw multiple tablet metadata entries for root table");
          TabletManagement tm = TabletManagementIterator.decode(e);
```

Reply:
Implemented in 85f67fe